src/backend/distributed/connection/remote_commands.c

```diff
@@ -716,14 +716,14 @@ PutRemoteCopyData(MultiConnection *connection, const char *buffer, int nbytes)
 	Assert(PQisnonblocking(pgConn));
 
 	int copyState = PQputCopyData(pgConn, buffer, nbytes);
-	if (copyState == -1)
+	if (copyState <= 0)
```

Comment:
this does not seem like a failure

From docs:
>  The result is 1 if the data was queued, zero if it was not queued because of full buffers (this will only happen in nonblocking mode),

however, the current code does seem suspect

in practice, we might always end up in FinishConnectionIO below (?), but maybe we should force it to go in there if copyState == 0

Reply:
> From docs:
> 
> > The result is 1 if the data was queued, zero if it was not queued because of full buffers (this will only happen in nonblocking mode),
> 
> this does not seem like a failure
> 
> however, the current code does seem suspect
> 
> in practice, we might always end up in FinishConnectionIO below (?), but maybe we should force it to go in there if copyState == 0

IMHO `FinishConnectionIO` may not help here, it try to send all pending data(which is hold in conn->outBuffer).
When copyState == 0 the data PQputCopyData try to send even not added to conn->outBuffer, we should either
treat this as a failure or try `PQputCopyData` again, but should not treat this as a success.

I see some usage in postgres upstream, all `PQputCopyData` are used with `<= 0`, the result 0 is returned when
failed to enlarge conn->outBuffer.